### PR TITLE
calibrado con eeprom

### PR DESCRIPTION
**Pull request by Albalate**
I have added an example to calibrate the scale and add the result in the eeprom memory of arduino UNO.
regards

### CODE REVIEW

**Comment by RobTillaart** [comment]
Thanks, good example.

 Some remarks:
- you changed two examples (pins only) to be able to run them on ESP32. Such change does not add anything to the example.  Better comment the original lines and add twe lines that states the changed pins are for ESP32. 
- the new example is a mix of English and Spanish. Although no problem I prefer to keep things English.
- the URL of the example points to a fork of the library. If it is merged into this library it should have an URL to the root repo (or no url). The risk is that issues will be filed at your fork of the lib which might lead to different versions in the future..

No showstoppers, will merge the PR in coming days.

**Comment by RobTillaart** [comment]
The build fails on  your new sketch with the eeprom code. It might be ESP specific.

**Comment by Albalate** [comment]
sorry, All of the following boards have an EEPROM: (https://docs.arduino.cc/learn/programming/eeprom-guide)
[Arduino UNO Rev.3](https://store.arduino.cc/products/arduino-uno-rev3/)
[Arduino UNO WiFi Rev.2](https://store.arduino.cc/arduino-uno-wiFi-rev2)
[Arduino Mega 2560 Rev.3](https://store.arduino.cc/products/arduino-mega-2560-rev3)
[Arduino Nano Every](https://store.arduino.cc/products/arduino-nano-every)
[Arduino Micro](https://store.arduino.cc/products/arduino-micro)
[Arduino Leonardo](https://store.arduino.cc/products/arduino-leonardo-with-headers)
[Arduino Nano](https://store.arduino.cc/products/arduino-nano)

EEPROM.h

**Comment by RobTillaart** [comment]
@Albalate 

check - https://github.com/RobTillaart/HX711/runs/5526296592?check_suite_focus=true

line 580 and more shows the errors during compilation of your example
```
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn arduino:avr:uno --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino: In function 'void StupEeprom()':
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:29:32: error: no matching function for call to 'EEPROMClass::begin(int)'
   if (!EEPROM.begin(EEPROM_SIZE))
                                ^
In file included from /home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:15:0:
/home/runner/.arduino15/packages/arduino/hardware/avr/1.8.5/libraries/EEPROM/src/EEPROM.h:125:11: note: candidate: EEPtr EEPROMClass::begin()
     EEPtr begin()                        { return 0x00; }
           ^~~~~
/home/runner/.arduino15/packages/arduino/hardware/avr/1.8.5/libraries/EEPROM/src/EEPROM.h:125:11: note:   candidate expects 0 arguments, 1 provided
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino: In function 'float SaveFloat(int, float)':
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:40:10: error: 'struct EEPROMClass' has no member named 'writeFloat'; did you mean 'write'?
   EEPROM.writeFloat(addr, param);//EEPROM.put(address, param);
          ^~~~~~~~~~
          write
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:41:10: error: 'struct EEPROMClass' has no member named 'commit'
   EEPROM.commit();
          ^~~~~~
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:42:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
 ^
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino: In function 'long int SaveLong(int, long int)':
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:44:10: error: 'struct EEPROMClass' has no member named 'writeFloat'; did you mean 'write'?
   EEPROM.writeFloat(addr, param);//EEPROM.put(address, param);
          ^~~~~~~~~~
          write
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:45:10: error: 'struct EEPROMClass' has no member named 'commit'
   EEPROM.commit();
          ^~~~~~
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:46:1: warning: no return statement in function returning non-void [-Wreturn-type]
 }
 ^
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino: In function 'void setup()':
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:123:12: error: 'struct EEPROMClass' has no member named 'commit'
     EEPROM.commit();
            ^~~~~~
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:125:5: error: 'ESP' was not declared in this scope
     ESP.restart();
     ^~~
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:125:5: note: suggested alternative: 'SP'
     ESP.restart();
     ^~~
     SP
Error during build: exit status 1
                       ✗
Compiling hHX_set_persistent.ino for adafruit:samd:adafruit_metro_m4... 
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn adafruit:samd:adafruit_metro_m4 --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:15:10: fatal error: EEPROM.h: No such file or directory
   15 | #include "EEPROM.h"
      |          ^~~~~~~~~~
compilation terminated.
Error during build: exit status 1
       ✗
```

----

You can restrict the platforms to only those that compile by adding a file to the example folder called **.arduino-ci.yml**

the content should be like this, but only the platforms that compile
```
compile:
  # Choosing to run compilation tests on 2 different Arduino platforms
  platforms:
    # - uno
    # - due
    # - zero
    # - leonardo
    # - m4
    - esp32
    # - esp8266
    # - mega2560
```
esp8266 might work too

Finally you must rename the example so it is EXPLICIT that it is an ESP32 specific example.

**Comment by RobTillaart** [comment]
Other patched example does fail to build too

```
Discovered example sketch...                                  HX_delta_scale.ino
Wildcard architecture in library.properties, using configured platforms... 
    uno
    m4
    esp32
...Wildcard architecture in library.properties, using configured platforms      
Board package arduino:avr has a defined URL...                                 ✓
Installing board package arduino:avr...                                        ✓
Board package esp32:esp32 has a defined URL...                                 ✓
Installing board package esp32:esp32...                                        ✓
Board package adafruit:samd has a defined URL...                               ✓
Installing board package adafruit:samd...                                      ✓
Compiling HX_delta_scale.ino for arduino:avr:uno... 
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn arduino:avr:uno --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino: In function 'void setup()':
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:15: error: 'dataPin' was not declared in this scope
   scale.begin(dataPin, clockPin);
               ^~~~~~~
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:15: note: suggested alternative: 'atan'
   scale.begin(dataPin, clockPin);
               ^~~~~~~
               atan
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:24: error: 'clockPin' was not declared in this scope
   scale.begin(dataPin, clockPin);
                        ^~~~~~~~
Error during build: exit status 1
                           ✗
Compiling HX_delta_scale.ino for adafruit:samd:adafruit_metro_m4... 
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn adafruit:samd:adafruit_metro_m4 --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino: In function 'void setup()':
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:15: error: 'dataPin' was not declared in this scope; did you mean 'atan'?
   26 |   scale.begin(dataPin, clockPin);
      |               ^~~~~~~
      |               atan
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:24: error: 'clockPin' was not declared in this scope
   26 |   scale.begin(dataPin, clockPin);
      |                        ^~~~~~~~
Error during build: exit status 1
           ✗
Compiling HX_delta_scale.ino for esp32:esp32:featheresp32:FlashFreq=80... 
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn esp32:esp32:featheresp32:FlashFreq=80 --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino: In function 'void setup()':
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:15: error: 'dataPin' was not declared in this scope
   scale.begin(dataPin, clockPin);
               ^~~~~~~
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:15: note: suggested alternative: 'atan'
   scale.begin(dataPin, clockPin);
               ^~~~~~~
               atan
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:24: error: 'clockPin' was not declared in this scope
   scale.begin(dataPin, clockPin);
                        ^~~~~~~~
/home/runner/Arduino/libraries/HX711/examples/HX_delta_scale/HX_delta_scale.ino:26:24: note: suggested alternative: 'clock_t'
   scale.begin(dataPin, clockPin);
                        ^~~~~~~~
                        clock_t
Error during build: exit status 1
     ✗
```

Missing declaration and initialization of the data and clock pin.

File: examples/HX_delta_scale/HX_delta_scale.ino [hunk 1]
```diff
@@ -11,8 +11,6 @@
 
 HX711 scale;
 
-uint8_t dataPin = 6;
-uint8_t clockPin = 7;
```
Comment (RobTillaart) [inline examples/HX_delta_scale/HX_delta_scale.ino:14]: MIssing declarations and initialization

File: examples/hHX_set_persistent/hHX_set_persistent.ino [hunk 2]
```diff
@@ -0,0 +1,138 @@
+
```
Comment (RobTillaart) [inline examples/hHX_set_persistent/hHX_set_persistent.ino:1]: need to add an .arduino-ci.yml file (see conversation)

File: HX711.h [hunk 3]
```diff
@@ -50,11 +50,11 @@ class HX711
...
-  //  raw read
-  float    read();
-  //  get average of multiple raw reads
-  //  times = 1 or more
-  float    read_average(uint8_t times = 10);
+
+  float    read();//  raw read
+
+
+  float    read_average(uint8_t times = 10);//  get average of multiple raw reads //  times = 1 or more
```
Comment (RobTillaart) [inline HX711.h:57]: What is the purpose of moving comments?
It does not improve the functionality or the readability (in my opinion).

**Comment by RobTillaart** [comment]
```
Compiling hHX_set_persistent.ino for arduino:avr:uno...                        ✓
Compiling hHX_set_persistent.ino for adafruit:samd:adafruit_metro_m4... 
Last command:  $  /home/runner/arduino-cli --format json compile --fqbn adafruit:samd:adafruit_metro_m4 --warnings all --dry-run /home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino
/home/runner/Arduino/libraries/HX711/examples/hHX_set_persistent/hHX_set_persistent.ino:11:10: fatal error: EEPROM.h: No such file or directory
   11 | #include "EEPROM.h"
      |          ^~~~~~~~~~
compilation terminated.
Error during build: exit status 1
       ✗
Compiling hHX_set_persistent.ino for esp32:esp32:featheresp32:FlashFreq=80...  ✓
```

File: examples/HX_delta_scale/HX_delta_scale.ino (hunk 1 above)
Comment (Albalate) [reply examples/HX_delta_scale/HX_delta_scale.ino]: is solved

File: examples/hHX_set_persistent/hHX_set_persistent.ino (hunk 2 above)
Comment (Albalate) [reply examples/hHX_set_persistent/hHX_set_persistent.ino:1]: the example is now for arduino uno

File: HX711.h (hunk 3 above)
Comment (Albalate) [reply HX711.h:57]: I move them for my convenience, they don't need to stay

**Comment by Albalate** [comment]
now complia for arduino uno, when I finish the project that I have in hand, I will give it more love to make it compatible with other architectures esp32 and esp8266

**Comment by RobTillaart** [comment]
> now complia for arduino uno, when I finish the project that I have in hand, I will give it more love to make it compatible with other architectures esp32 and esp8266

Thanks,

File: HX711.h (hunk 3 above)
Comment (RobTillaart) [reply HX711.h:57]: Every person has its own preferred style of indenting, comments etc. Changing these, especially if done partially, disrupt the structure the maintainer of the code uses. This makes it harder for the maintainer to go through the code. As I try to keep all my 150++ libraries somehow consistent in terms of style - which is hard enough - I will not merge any PR that makes changes in comments or layout or style without a very good reason.

So please revert these changes.

**Comment by Albalate** [comment]
the EEPROM.h library is included in the standard arduino libraries, I have compiled with Atom, Vsc, and the arduino ide v1 and v2, without any problem. the test is not correct.
regards

**Comment by RobTillaart** [comment]
> the EEPROM.h library is included in the standard arduino libraries, I have compiled with Atom, Vsc, and the arduino ide v1 and v2, without any problem. the test is not correct. regards

You need to add an .arduino-ci.yml file, I mentioned before

**Comment by RobTillaart** [comment]
@Albalate 
Check - https://github.com/RobTillaart/Cozir/tree/master/examples/Cozir_MEGA_3_channel_CO2
it has a **.arduino-CI.yml** to only allow a MEGA to run, the others platforms are commented.

You need to make a file that only allows UNO and ESP32 and comment the others.

Then it will compile and I can merge

**Comment by Albalate** [comment]
> > the EEPROM.h library is included in the standard arduino libraries, I have compiled with Atom, Vsc, and the arduino ide v1 and v2, without any problem. the test is not correct. regards
> 
> You need to add an .arduino-ci.yml file, I mentioned before

I already had the file with only one, esp and m4.
now I have tried only with, UNO

**Comment by RobTillaart** [comment]
And there is green light!

Can you please enable the ESP32 too in the .arduino-CI.yml file as that one compiled too.

**Comment by RobTillaart** [comment]
It looks perfect, will merge tomorrow

**Pull request merged**